code/datums/components/tackle.dm

```diff
@@ -0,0 +1,338 @@
+			target.Knockdown(20)
+
+		if(3 to 4) // really good hit, the target is definitely worse off here. Without positive modifiers, this is as good a tackle as you can land
+			user.visible_message("<span class='warning'>[user] lands an expert tackle on [target], knocking [target.p_them()] down hard!</span>", "<span class='userdanger'>You land an expert tackle on [target], knocking [target.p_them()] down hard!</span>", target)
+			to_chat(target, "<span class='userdanger'>[user] lands an expert tackle on you, knocking you down hard!</span>")
+
+			target.take_bodypart_damage(stamina=40)
+			target.Paralyze(5)
+			user.Knockdown(10)
+			target.Knockdown(30)
+
+		if(5 to INFINITY) // absolutely BODIED
+			user.visible_message("<span class='warning'>[user] lands a monster tackle on [target], knocking [target.p_them()] senseless and applying an aggressive pin!</span>", "<span class='userdanger'>You land a monster tackle on [target], knocking [target.p_them()] senseless and applying an aggressive pin!</span>", target)
+			to_chat(target, "<span class='userdanger'>[user] lands a monster tackle on you, knocking you senseless and aggressively pinning you!</span>")
+
+			user.SetKnockdown(0)
+			target.take_bodypart_damage(stamina=40)
+			if(ishuman(target) && ishuman(user))
+				S.dna.species.grab(S, T)
+				S.setGrabState(GRAB_AGGRESSIVE)
+				target.Paralyze(15)
+				target.Knockdown(30)
+
+	return COMPONENT_MOVABLE_IMPACT_FLIP_GENTLE
+
+
+/*
+	rollTackle() handles all of the modifiers for the actual carbon-on-carbon tackling, and gets its own proc because of how many there are (with plenty more in mind!)
+
+	The base roll is between (-3, 3), with negative numbers favoring the target, and positive numbers favoring the tackler. The target and the tackler are both assessed for
+		how easy they are to knock over, with clumsiness and dwarfiness being strong maluses for each, and gigantism giving a bonus for each. These numbers and ideas
+		are absolutely subject to change.
+
+	In addition, after subtracting the defender's mod and adding the attacker's mod to the roll, the component's base (skill) mod is added as well. Some sources of tackles
+		are better at taking people down, like the bruiser and rocket gloves, while the dolphin gloves have a malus in exchange for better mobility.
+
+*/
+/datum/component/tackler/proc/rollTackle(mob/living/carbon/target)
+	var/defense_mod = 0
+	var/attack_mod = 0
+
+	// DE-FENSE
+	if(target.drunkenness > 60) // drunks are easier to knock off balance
+		defense_mod -= 3
+	else if(target.drunkenness > 30)
+		defense_mod -= 1
+	if(HAS_TRAIT(target, TRAIT_CLUMSY))
+		defense_mod -= 2
+	if(HAS_TRAIT(target, TRAIT_FAT)) // chonkers are harder to knock over
+		defense_mod += 1
+	if(HAS_TRAIT(target, TRAIT_GRABWEAKNESS))
+		defense_mod -= 2
+
+	if(ishuman(target))
+		var/mob/living/carbon/human/T = target
+		if(T.dna.check_mutation(DWARFISM))
+			defense_mod -= 2
+		if(T.dna.check_mutation(GIGANTISM))
+			defense_mod += 2
```

Comment:
probably best if these checks use traits, which can be made part of these mutations

Reply:
agreed